src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useContext, useRef, useEffect } from 'react';
+import { DispatchContext, TodosContext } from '../../Store';
+import { Todo } from '../../Types/Todo';
+
+type Props = {
+  title: string
+  completed: boolean,
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ title, completed, todo }) => {
+  const dispatch = useContext(DispatchContext);
+  const { edit, newTitle } = useContext(TodosContext);
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const handlerChange = () => {
+    dispatch({ type: 'todoCompleted', payLoad: todo });
+  };
+
+  const handleKeyUp = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter' && !event.target.value) {
+      dispatch({ type: 'removeTodo', payLoad: todo });
+    }
+
+    if (event.key === 'Enter') {
+      dispatch({ type: 'saveNewTitleTodo', payLoad: todo });
+    }
+
+    if (event.key === 'Escape') {
+      dispatch({ type: 'removeEdit' });
+    }
+  };
+
+  useEffect(() => {
+    if (edit) {
+      titleField.current?.focus();
+    }
+  }, [edit]);
+
+  return (
+    <li
+      className={`${edit === todo.id && 'editing'} ${completed && 'completed'}`}
+    >
+      <div className="view">
+        <input
+          type="checkbox"
+          className="toggle"
+          id="toggle-view"
+          onChange={handlerChange}
+        />
+        <label
+          htmlFor="toggle-view"
+          onDoubleClick={() => {
+            dispatch({ type: 'edit', payLoad: todo });
+          }}
+        >
+          {title}
+        </label>
+
+        <button
+          aria-label={title}
+          type="button"
+          className="destroy"
+          data-cy="deleteTodo"
+          onClick={() => dispatch({ type: 'removeTodo', payLoad: todo })}
+        />
+      </div>
+      <input
+        ref={titleField}
+        value={newTitle}
+        type="text"
+        className="edit"
+        onChange={(event) => dispatch({
+          type: 'changeTitleTodo',
```

Comment:
You are dispatching the `changeTitleTodo` action on every change event in the input field. This could lead to unnecessary re-renders of the component. Consider debouncing this action or dispatching it only when the input field loses focus.